users/views.py

```diff
@@ -10,42 +10,60 @@
 from rest_framework_simplejwt.tokens import RefreshToken
 from urllib.parse import urlencode
 import json
+import base64
+
 
 class CustomUserCreateView(APIView):
     permission_classes = (AllowAny,)
 
     def post(self, request, *args, **kwargs):
-        serializer = CustomUserSerializer(data=request.data)
+        serializer = CustomUserSerializer(data=request.data, context={'request': request})
         serializer.is_valid(raise_exception=True)
         
-        user_data = serializer.validated_data
-        encoded_user_data = urlencode(user_data)
+        custom_user_data = serializer.validated_data
+        user = serializer.save()

```

Comment:
don't save the user in this API, send all the User's data on email in encoded form.
Write the logic in the VerifyEmailView to encode them and save them in the DB.